src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -27,54 +25,38 @@ internal sealed partial class CSharpSelectionValidator(
     {
         private readonly bool _localFunction = localFunction;
 
-        protected override bool AreStatementsInSameContainer(StatementSyntax statement1, StatementSyntax statement2)
+        protected override SelectionInfo UpdateSelectionInfo(
+            SelectionInfo selectionInfo, StatementSyntax? firstStatement, StatementSyntax? lastStatement, CancellationToken cancellationToken)
         {
-            if (statement1.Parent == statement2.Parent)
-                return true;
-
-            if (statement1.Parent is GlobalStatementSyntax
-                && statement2.Parent is GlobalStatementSyntax
-                && statement1.Parent.Parent == statement2.Parent.Parent)
-            {
-                return true;
-            }
-
-            return false;
-        }
-
-        protected override SelectionInfo GetInitialSelectionInfo(CancellationToken cancellationToken)
-        {
-            var text = SemanticDocument.Text;
             var root = SemanticDocument.Root;
             var model = SemanticDocument.SemanticModel;
 
             // go through pipe line and calculate information about the user selection
-            var selectionInfo = GetInitialSelectionInfo(root, text);
-            selectionInfo = AssignInitialFinalTokens(selectionInfo, root, cancellationToken);
+            selectionInfo = AssignInitialFinalTokens(selectionInfo, firstStatement, lastStatement);
             selectionInfo = AdjustFinalTokensBasedOnContext(selectionInfo, model, cancellationToken);
-            selectionInfo = AssignFinalSpan(selectionInfo, text);
-            selectionInfo = ApplySpecialCases(selectionInfo, text, SemanticDocument.SyntaxTree.Options, _localFunction);
```

Comment:
we used to passa a lot of instance-state data to helper instance methods.  i cleaned that up to access off of 'this' instead.